src/WatchdogLogger.php

```diff
@@ -82,7 +82,96 @@ public function log($level, $message, array $context = array())
+     *   The message to store in the log. If empty, a text that contains all useful
+     *   information about the passed-in exception is used.
+     * @param array $variables
+     *   Array of variables to replace in the message on display. Defaults to the
+     *   return value of _drupal_decode_exception().
+     * @param int $severity
+     *   The severity of the message, as per RFC 3164.
+     * @param string $link
+     *   A link to associate with the message.
+     *
+     */
+    protected function logThrowable($type, $exception, $message = null, $variables = [], $severity = WATCHDOG_ERROR, $link = null)
+    {
+        // Use a default value if $message is not set.
+        if (empty($message)) {
+            // The exception message is run through check_plain() by _drupal_decode_exception().
+            $message = '%type: !message in %function (line %line of %file).';
+        }
+
+        // $variables must be an array so that we can add the exception information.
+        if (!is_array($variables)) {
+            $variables = array();
+        }
+
+        require_once DRUPAL_ROOT . '/includes/errors.inc';
+        $variables += _drupal_decode_exception($exception);
+
+        $this->watchdog($type, $message, $variables, $severity, $link);
+    }
+
+
+    /**
+     * Logs a message using Watchdog.
+     *
+     * This is mostly just a wrapper for the watchdog() function.
+     *
+     * @param string $type
+     *   The category to which this message belongs. Can be any string, but the
+     *   general practice is to use the name of the module calling watchdog().
+     * @param string $message
+     *   The message to store in the log. Keep $message translatable
+     *   by not concatenating dynamic values into it! Variables in the
+     *   message should be added by using placeholder strings alongside
+     *   the variables argument to declare the value of the placeholders.
+     *   See t() for documentation on how $message and $variables interact.
+     * @param array $variables
+     *   Array of variables to replace in the message on display or
+     *   NULL if message is already translated or not possible to
+     *   translate.
+     * @param int $severity
+     *   The severity of the message; one of the following values as defined in
+     *   RFC 3164.
+     *   - WATCHDOG_EMERGENCY: Emergency, system is unusable.
+     *   - WATCHDOG_ALERT: Alert, action must be taken immediately.
+     *   - WATCHDOG_CRITICAL: Critical conditions.
+     *   - WATCHDOG_ERROR: Error conditions.
+     *   - WATCHDOG_WARNING: Warning conditions.
+     *   - WATCHDOG_NOTICE: (default) Normal but significant conditions.
+     *   - WATCHDOG_INFO: Informational messages.
+     *   - WATCHDOG_DEBUG: Debug-level messages.
+     * @param string $link
+     *   A link to associate with the message.
+     *
+     */
+    protected function watchdog($type, $message, $variables = [], $severity = WATCHDOG_NOTICE, $link = null)
```

Comment:
I'm not sure we're gaining all that much by wrapping this instead of using direct calls.